### PR TITLE
mod: add sha2 built-in function

### PR DESCRIPTION
**Pull request by zs634134578**
sha2 built-in function

### CODE REVIEW

**Comment by shenli** [comment]
@zs634134578 Thanks! Please fix ci.

File: expression/builtin_encryption.go [hunk 1]
```diff
@@ -580,9 +583,58 @@ type builtinSHA2Sig struct {
...
+)
+
 // See https://dev.mysql.com/doc/refman/5.7/en/encryption-functions.html#function_sha2
 func (b *builtinSHA2Sig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("SHA2")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	// SHA2 function accepts 2 parameters
```
Comment (XuHuaiyu) [inline expression/builtin_encryption.go]: Length of arguments would be checked in `verifyArgs` before,
so we don't need to check it here.

File: expression/builtin_encryption.go [hunk 2]
```diff
@@ -580,9 +583,58 @@ type builtinSHA2Sig struct {
 	baseBuiltinFunc
 }
 
+// Supported hash length as the second argument
```
Comment (XuHuaiyu) [inline expression/builtin_encryption.go]: Supported hash length of SHA-2 family.

File: expression/builtin_encryption_test.go [hunk 3]
```diff
@@ -132,6 +132,65 @@ func (s *testEvaluatorSuite) TestShaEncrypt(c *C) {
...
+		c.Assert(err, IsNil)
+		if test.validCase {
+			res, err := crypt.ToString()
+			c.Assert(err, IsNil)
+			c.Assert(res, Equals, test.crypt)
+		} else {
+			c.Assert(crypt.IsNull(), IsTrue)
+		}
+	}
+	// Corner case 1: NULL input for sha2
```
Comment (XuHuaiyu) [inline expression/builtin_encryption_test.go]: For 0 args, and other incorrect parameter count, MySQL will cause error:
`ERROR 1582 (42000): Incorrect parameter count in the call to native function 'sha2'`

TiDB will check the count of parameters [here](https://github.com/pingcap/tidb/blob/master/expression/builtin.go#L109)

**Comment by XuHuaiyu** [comment]
LGTM

**Comment by coocood** [comment]
LGTM
Thank you for your contribution.
